addons/enemybar/enemybar.lua

```diff
@@ -24,31 +24,524 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+windower.register_event('incoming chunk', handle_action_packet)
+windower.register_event('zone change', reset_tracked_actions)
+windower.register_event('addon command', handle_command)
+windower.register_event('logout', function(...)
+    player_id = nil    
+    state = {}      
+end)
+windower.register_event('login', function(...)
     if windower.ffxi.get_info().logged_in then
         player_id = windower.ffxi.get_player().id
+        load_settings()
     end
+    state = {}
 end)
 
-windower.register_event('prerender', render_target_bar)
-windower.register_event('prerender', render_subtarget_bar)
-windower.register_event('target change', target_change)
+-- Handle drag and drop
+windower.register_event('mouse', function(type, x, y, delta, blocked)
+    if blocked or not bar_sets then
+        return
+    end
 
-windower.register_event('logout', function(...)
-    -- This is a super cheap fix, but it works. 
-    windower.send_command("input //lua r enemybar");        
+    -- Mouse drag
+    if type == 0 then
+        if dragged then
+            local d_x = 0
+            local d_y = 0
+            if drag_snap_key_down then
+                -- snap
+                d_x = math.floor(((x - dragged.x) + dragged.bars[1].x)/10)*10 - dragged.bars[1].x
+                d_y = math.floor(((y - dragged.y) + dragged.bars[1].y)/10)*10 - dragged.bars[1].y
+            else
+                -- no snap
+                d_x = x - dragged.x
+                d_y = y - dragged.y
+            end
+
+            for i, b in ipairs(dragged.bars) do
+                    bars.move(b, b.x + d_x, b.y + d_y)
+            end
+            dragged.x = dragged.x + d_x
+            dragged.y = dragged.y + d_y
+            return true
+        end
+
+    -- Mouse left click
+    elseif type == 1 then
+        if not state.setup then return false end
+        for _, s in ipairs(bar_sets) do
+            for _, b in ipairs(s) do
+                if bars.hover(b, x, y) then
+                    dragged = {bars = s, x = x , y = y}
+                    return true
+                end
+            end
+        end
+
+    -- Mouse left release
+    elseif type == 2 then
+        if dragged then
+            settings.target_bar.pos = {x=target_bar.x,y=target_bar.y}
+            settings.subtarget_bar.pos = {x=subtarget_bar.x,y=subtarget_bar.y}
+            settings.focustarget_bar.pos = {x=focustarget_bar.x,y=focustarget_bar.y}
+            settings.aggro_bar.pos = {x=aggro_bars[1].x,y=aggro_bars[1].y}
+            settings:save()
+            dragged = nil
+            return true
+        end
+    end
+
+    return false
 end)
+windower.register_event('keyboard', function(dik, down) -- lol diks
+    if dik == 29 then -- if ctrl
+        drag_snap_key_down = down
+        --windower.add_to_chat(1,'ctrl: '..tostring(down))
+    end
+end)
+
+if windower.ffxi.get_info().logged_in then
+    player_id = windower.ffxi.get_player().id
+    load_settings()
```

Comment:
You should not need to call this, either here or in the `login`/`logout` events. See my comment above at `load_defaults`.